### ISSUE TITLE
Simplify karma dependency injection.  

### ISSUE THREAD

**Issue by johnjbarton** [karma-runner/karma#3533]
The current system is too fancy and mysterious. DI is by its nature 'mysterious' because the values are set in code not directly known to the getter.  But karma system compounds this mystery with the auto-annotation feature and the ability to do the same thing in multiple ways.

**Comment by devoto13** [comment]
What is your vision here?

I've been thinking about unifying on `.$inject = [token1, token2, token3]` (e.g. [this](https://github.com/karma-runner/karma/blob/master/lib/file-list.js#L240)) and exporting and documenting constants for all known DI tokens (e.g. [these](https://github.com/karma-runner/karma/blob/master/lib/server.js#L70)).

**Comment by johnjbarton** [comment]
Yes, unifying on that form. But also deprecating and removing the function parsing thing
https://github.com/vojtajina/node-di/blob/master/lib/annotation.js
